src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,135 @@
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new DukeException("The description of a " + type + " cannot be empty");
+        }
+    }
+
+    /**
+     * Executes the AddCommand, adds the task to the LinkedList and saves it to the file.
+     *
+     * @param tasks   The TaskList containing all existing tasks.
+     * @param ui      The Ui for printing purposes.
+     * @param storage The Storage for saving tasks to file.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+
+        tasks.allTasks.add(task);
+        StringBuilder sb = new StringBuilder("Got it. I've added this task:\n" + task);
+        if (tasks.allTasks.size() == 1) {
+            sb.append("\nNow you have ")
+              .append(tasks.allTasks.size())
+                .append(" task in the list.");
+        } else {
+            sb.append("\nNow you have ")
+              .append(tasks.allTasks.size())
+                .append(" tasks in the list.");
+        }
+        storage.appendTaskToFile(task);
+        ui.printMessage(sb.toString());
+    }
+
+    /**
+     * Checks if it is a bye command.
+     *
+     * @return A boolean: true if it is a bye command.
+     */
+    public boolean isExit() {
+        return false;
+    }
+
+    /**
+     * Parses a string and reformats it as a date.
+     *
+     * @return The string representation of the new date object.
+     */
+    public String parseDate() {
+        try {
+            SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy HHmm", java.util.Locale.ENGLISH);
+            Date myDate = sdf.parse(time.replaceAll("-", "/"));
+            sdf.applyPattern("EEE, d MMM yyyy, hh:mm a");
+            return sdf.format(myDate);
+        } catch (ParseException e) {
+            return time.trim();
+        }
+    }
+
+    /**
+     * Parses the command description for to-dos and checks if it is empty/null.
+     *
+     * @return The parsed description only if it is not empty/null.
+     */
+    public String parseDescription() {
+        if (description.length() < 1) {
+            throw new ArrayIndexOutOfBoundsException();
+        } else {
+            return description;
+        }
+    }
+
+    /**
+     * Parses the command description with delimiters for events and deadlines and checks if it is in
+     * the correct format.
+     *
+     * @param delimiter The delimiter that splits the description.
+     * @return The parsed description.
+     * @throws DukeException In the event that the command is in the wrong format.
+     */
+    public String parseDescription(String delimiter) throws DukeException {
+        String[] details = description.split(delimiter);
+        if (details.length == 1) {
+            if (details[0].length() != 0) {
+                throw new DukeException("Incorrect format. \nPlease try again with the format below: \n"
+                        + type + " ($task_name) " + delimiter + " ($date/day)");
+            }
```

Comment:
Very considerate developer in telling user to input the correct format!

Reply:
You may also want to consider extending different errors from `DukeException`. This would help make the code more readable and minimize the many '+' when trying to build your string. you can simply create a new `DukeException` and passing the arguments `$(task_name)` and `$(date/day`) as parameters